medusa/storage/s3_base_storage.py

```diff
@@ -136,7 +136,8 @@ def connect(self):
             region_name=self.credentials.region,
             signature_version='v4',
             tcp_keepalive=True,
-            max_pool_connections=max_pool_size
+            max_pool_connections=max_pool_size,
+            read_timeout=int(self.config.read_timeout),
```

Comment:
**Question:** is there a similar feature for GCP and Azure? It would be nice to apply the same settings for all 3.

Reply:
@rzvoncek ?

Reply:
Yes, they are available. I've added commits applying the read timeout config value to Azure and GCP read operations.